clang/lib/DPCT/GenMakefile.cpp

```diff
@@ -77,26 +77,39 @@ static void getCompileInfo(
     std::string FileName = Entry.first;
     if (llvm::StringRef(FileName).startswith("LinkerEntry")) {
       // Parse linker cmd to get target name and objfile names
-      std::unordered_map<std::string, bool> ObjsInLinker;
+      std::vector<std::string> ObjsInLKOrARCmd;
 
       bool IsTargetName = false;
+      bool IsArCommand = false;
+      bool SkipArOptions = false;
+
       std::string ObjName;
       std::string TargetName;
+      std::string Tool;
       for (const auto &Obj : Entry.second) {
         if (Obj == "-o") {
           IsTargetName = true;
         } else if (IsTargetName) {
           // Set the target name
           TargetName = Obj;
           IsTargetName = false;
+          Tool = "ld -o";
         } else if (llvm::StringRef(Obj).endswith(".o")) {
           llvm::SmallString<512> FilePathAbs(Obj);
           llvm::sys::path::native(FilePathAbs);
           llvm::sys::fs::make_absolute(FilePathAbs);
           llvm::sys::path::remove_dots(FilePathAbs, true);
-          ObjsInLinkerCmd[std::string(FilePathAbs.str())] = true;
-          ObjsInLinker[std::string(FilePathAbs.str())] = true;
+          ObjsInLKOrARCmd.push_back(std::string(FilePathAbs.str()));
           ObjName = std::string(FilePathAbs.str());
+        } else if (Obj == "ar") {
```

Comment:
could you add comment for the new added logic.

Reply:
okay